crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -151,7 +191,26 @@ impl AuthorityStorePruner {
                         } else {
                             error!("Failed to flush objects table");
                         }
-                    }
+                    },
+                    Some((state, callback)) = checkpoint_stream.recv() => {
+                        if !enable_live_pruner {
```

Comment:
it was the bug: disabled pruner still needs to send callback to checkpoint executor